src/pages/experiments/[experimentId]/data-processing/data-integration/components/CalculationConfig.jsx

```diff
@@ -157,21 +143,23 @@ const CalculationConfig = () => {
         <div style={{ paddingLeft: '1rem' }}>
           <Form.Item label='Number of Principal Components'>
             <InputNumber
-              value={settings.dimensionalityReduction.numPCs}
-              onChange={(value) => updateSettings({ dimensionalityReduction: { numPCs: value } })}
+              value={numPCs}
+              onChange={(value) => setNumPCs(value)}
+              onPressEnter={(e) => e.preventDefault()}
               onStep={(value) => updateSettings({ dimensionalityReduction: { numPCs: value } })}
+              onBlur={(e) => updateSettings({ dimensionalityReduction: { numPCs: parseInt(e.target.value, 0) } })}
             />
```

Comment:
Used a local state (`numPCs`) to buffer changes in the UI, and only fire updates to store when finished changing (`onBlur`, `onStep`). `onPressEnter` is disabled to prevent triggering of `Apply` button when user presses enter in this field.